File: weha/source/weha/diff/Match.php
<?php
include_once('BasicEdit.php');

class Match extends BasicEdit {

	private $length;

	private $oldOrder;
	private $newOrder;
	
	public function getLength() {
		return $this->length;
	}

	public function getOldPos() {
		return $this->oldPos;
	}

	public function getNewPos() {
		return $this->newPos;
	}
	
	public function getOldOrder() {
		return $this->oldOrder;
	}

	public function setOldOrder($oldOrder) {
		$this->oldOrder = $oldOrder;
	}

	public function getNewOrder() {
		return $this->newOrder;
	}

	public function setNewOrder($newOrder) {
		$this->newOrder = $newOrder;
	}

	public function __construct($length, $oldPos, $newPos) {
		$this->length = $length;
		$this->oldPos = $oldPos;
		$this->newPos = $newPos;
		$this->oldTokens = null;
		$this->newTokens = null;
	}

	public function getDescription() {
		return sprintf("Match(%d, %d, %d)", $this->length, $this->oldPos, $this->newPos);
	}

	public function getContent() {
		return null;
	}

}

File: weha/source/weha/wikianalysis/diff/token/Match.java
package mo.umac.wikianalysis.diff.token;

import mo.umac.wikianalysis.lexer.WikiToken;

public class Match extends BasicEdit {

	protected int length;

	private int oldOrder;
	private int newOrder;
	
	public int getLength() {
		return length;
	}

	public int getOldPos() {
		return oldPos;
	}

	public int getNewPos() {
		return newPos;
	}
	
	public int getOldOrder() {
		return oldOrder;
	}

	public void setOldOrder(int oldOrder) {
		this.oldOrder = oldOrder;
	}

	public int getNewOrder() {
		return newOrder;
	}

	public void setNewOrder(int newOrder) {
		this.newOrder = newOrder;
	}

	public Match(int length, int oldPos, int newPos) {
		super();
		this.length = length;
		this.oldPos = oldPos;
		this.newPos = newPos;
		this.oldTokens = null;
		this.newTokens = null;
	}

	public String getDescription() {
		return String.format("Match(%d, %d, %d)", length, oldPos, newPos);
	}

	@Override
	public WikiToken[] getContent() {
		return null;
	}

	@Override
	public String getLinkedDesc() {
		return String.format("Match(%d, <a href='#o%d'>%d</a>, <a href='#n%d'>%d</a>)", length, oldPos, oldPos, newPos, newPos);
	}

}
